mempool/src/pool/mod.rs

```diff
@@ -975,6 +976,10 @@ impl<M: MemoryUsageEstimator> Mempool<M> {
         let event = event::NewTip::new(block_id, block_height);
         self.events_controller.broadcast(event.into());
     }
+
+    pub fn current_fee_rate(&self) -> u128 {
+        self.rolling_fee_rate.read().rolling_minimum_fee_rate().atoms_per_kb()
+    }
```

Comment:
Please note that minimum fee rate is the minimum fee rate to get the transaction into mempool. It's probably not a good idea to use this for transaction fee rate if there should be any hope for the transaction to be included soon. If mempool is close to being full, chances of the transaction of this feerate being evicted are pretty high. A better number is something like the fee rate that places you N megabytes from the top, where N is a user-tunable parameter.

Reply:
Changed the algorithm, if you can check it out if you think it is ok